addons/payment/tests/test_payment_acquirer.py

```diff
@@ -0,0 +1,66 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import tagged
+
+from odoo.addons.payment.tests.common import PaymentCommon
+
+
+@tagged('-at_install', 'post_install')
+class TestPaymentAcquirer(PaymentCommon):
+
+    def test_unpublished_acquirer_not_compatible_for_public_user(self):
+        self.acquirer.published = False
+        self.env = self.env(user=self.env.ref('base.public_user'))
+
+        acquirers_public_user = self.env['payment.acquirer'].sudo()._get_compatible_acquirers(
+            company_id=self.company.id,
+            partner_id=self.partner.id,
+        )
+        self.assertNotIn(self.acquirer, acquirers_public_user)
+
+    def test_unpublished_acquirer_not_compatible_for_portal_user(self):
+        self.acquirer.published = False
+        self.env = self.env(user=self.portal_user)
+
+        acquirers_portal_user = self.env['payment.acquirer'].sudo()._get_compatible_acquirers(
+            company_id=self.company.id,
+            partner_id=self.partner.id,
+        )
+        self.assertNotIn(self.acquirer, acquirers_portal_user)
+
+    def test_unpublished_acquirer_compatible_for_internal_user(self):
+        self.acquirer.published = False
+
+        aquirers_internal_user = self.env['payment.acquirer']._get_compatible_acquirers(
+            company_id=self.company.id,
+            partner_id=self.partner.id,
+        )
+        self.assertIn(self.acquirer, aquirers_internal_user)
+
+    def test_published_acquirer_compatible_for_public_user(self):
+        # The acquirer's published field is set to True in SetUp
+        self.env = self.env(user=self.env.ref('base.public_user'))
+
+        acquirers_public_user = self.env['payment.acquirer'].sudo()._get_compatible_acquirers(
+            company_id=self.company.id,
+            partner_id=self.partner.id,
+        )
+        self.assertIn(self.acquirer, acquirers_public_user)
+
+    def test_published_acquirer_compatible_for_portal_user(self):
+        # The acquirer's published field is set to True in SetUp
+        self.env = self.env(user=self.portal_user)
+
+        acquirers_portal_user = self.env['payment.acquirer'].sudo()._get_compatible_acquirers(
+            company_id=self.company.id,
+            partner_id=self.partner.id,
+        )
+        self.assertIn(self.acquirer, acquirers_portal_user)
+
+    def test_published_acquirer_compatible_for_internal_user(self):
+        # The acquirer's published field is set to True in SetUp
+        aquirers_internal_user = self.env['payment.acquirer']._get_compatible_acquirers(
+            company_id=self.company.id,
+            partner_id=self.partner.id,
+        )
+        self.assertIn(self.acquirer, aquirers_internal_user)
```

Comment:
Same as for the previous comments:
- These tests are essentially the same, so they can be merged together.
- The resulting test is the most straightforward flow so it should be moved on the top of the file.